src/app/search/map-friendly-values.pipe.ts

```diff
@@ -194,11 +132,47 @@ export class MapFriendlyValuesPipe implements PipeTransform {
     }
     // Handle number
     const subBucketString: string = subBucket.toString();
-    // Handle string
-    if (this.friendlyValueNames.has(key) && this.friendlyValueNames.get(key).get(subBucketString)) {
-      return this.friendlyValueNames.get(key).get(subBucketString);
-    } else {
-      return subBucketString;
+
+    switch (key) {
+      case 'descriptorType':
+      case 'descriptor_type':
+        const shortFriendlyName =
+          this.descriptorLanguageService.descriptorLanguageBeanValueToExtendedDescriptorLanguageBean(subBucketString).shortFriendlyName;
+        if (shortFriendlyName) {
```

Comment:
For future reference, the "null coalescing" operator `??` is pretty handy for situations like this (much more compact and hard to modify incorrectly), ala:

`return shortFriendlyName ?? subBucketString;`

A bit further down, there might be opportunities to also use the "safe navigation" operator `?.`, for example:

`return fileTypes?.tabName ?? subBucketString`

Reply:
Thanks, I used these suggestions